include/istio/utils/local_attributes.h

```diff
@@ -0,0 +1,44 @@
+/* Copyright 2018 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ISTIO_UTILS_LOCAL_ATTRIBUTES_H
+#define ISTIO_UTILS_LOCAL_ATTRIBUTES_H
+
+#include "mixer/v1/attributes.pb.h"
+
+using ::istio::mixer::v1::Attributes;
+
+namespace istio {
+namespace utils {
+
+struct LocalAttributes {
+  LocalAttributes(const Attributes& inbound, const Attributes& outbound,
+                  const Attributes& forward)
+      : inbound(inbound), outbound(outbound), forward(forward) {}
+
+  // local inbound attributes
+  const Attributes inbound;
+
+  // local outbound attributes
+  const Attributes outbound;
+
+  // local forward attributes
```

Comment:
I think we should just have a map from a string to a service config in the bootstrap (https://github.com/istio/api/blob/master/mixer/v1/config/client/client_config.proto#L52), and then add a string reference to service configs in LDS for defaulting. I don't like inbound/outbound stuff here, since it's not applicable to middle proxies.